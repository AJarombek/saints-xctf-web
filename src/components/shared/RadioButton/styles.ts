/**
 * JSS styles for the RadioButton component.
 * @author Andrew Jarombek
 * @since 11/19/2020
 */

import Colors from '../../../styles/colors';
import { FontMixins } from '../../../styles/mixins';

export default {
  radio: {
    '& > label': {
      display: 'flex',
      alignItems: 'center'
    }
  },
  input: {
    opacity: 0,
    width: 0,
    height: 0
  },
  customRadio: {
    width: 16,
    height: 16,
    borderRadius: '50%',
    border: `2px solid ${Colors.sxctfRed}`,
    cursor: 'pointer'
  },
  customLabel: {
    ...FontMixins.roboto(),
    marginLeft: 10
  }
};
