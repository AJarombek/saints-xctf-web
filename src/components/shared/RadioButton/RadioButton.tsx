/**
 * Component which creates a custom radio button.
 * @author Andrew Jarombek
 * @since 11/19/2020
 */

import React, { memo } from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import classNames from 'classnames';
import { ClassValue } from 'classnames/types';

interface Props {
  id: string;
  name: string;
  value: string;
  label: React.ReactNode;
  defaultChecked?: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  className?: ClassValue;
}

const useStyles = createUseStyles(styles);

const RadioButton: React.FunctionComponent<Props> = ({
  id,
  name,
  value,
  label,
  defaultChecked = false,
  onChange,
  className
}) => {
  const classes = useStyles();

  return (
    <div className={classNames(classes.radio, className)}>
      <label>
        <div className={classes.inputWrapper}>
          <input
            type="radio"
            id={id}
            name={name}
            value={value}
            className={classes.input}
            defaultChecked={defaultChecked}
            onChange={onChange}
          />
          <div className={classes.customRadio} />
        </div>
        <div className={classes.customLabel}>{label}</div>
      </label>
    </div>
  );
};

export default memo<Props>(RadioButton);
