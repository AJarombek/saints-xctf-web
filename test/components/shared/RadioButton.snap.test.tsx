/**
 * Snapshot test for the RadioButton component.
 * @author Andrew Jarombek
 * @since 3/8/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from '../../../src/components/shared/RadioButton/RadioButton';

describe('RadioButton Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <RadioButton
            id="sunday"
            name="weekStart"
            value="sunday"
            label="Sunday"
            onChange={(): void => null}
            defaultChecked={true}
            className="customClass1"
          />
          <RadioButton
            id="monday"
            name="weekStart"
            value="monday"
            label="Monday"
            onChange={(): void => null}
            defaultChecked={false}
            className="customClass2"
          />
        </>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
