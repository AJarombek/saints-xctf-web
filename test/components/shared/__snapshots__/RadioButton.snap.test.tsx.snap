// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Snapshot Tests renders correctly 1`] = `
Array [
  <div
    className="radio-0-1-1 customClass1"
  >
    <label>
      <div
        className="inputWrapper-0-1-2"
      >
        <input
          className="input-0-1-3"
          defaultChecked={true}
          id="sunday"
          name="weekStart"
          onChange={[Function]}
          type="radio"
          value="sunday"
        />
        <div
          className="customRadio-0-1-4"
        />
      </div>
      <div
        className="customLabel-0-1-5"
      >
        Sunday
      </div>
    </label>
  </div>,
  <div
    className="radio-0-1-1 customClass2"
  >
    <label>
      <div
        className="inputWrapper-0-1-2"
      >
        <input
          className="input-0-1-3"
          defaultChecked={false}
          id="monday"
          name="weekStart"
          onChange={[Function]}
          type="radio"
          value="monday"
        />
        <div
          className="customRadio-0-1-4"
        />
      </div>
      <div
        className="customLabel-0-1-5"
      >
        Monday
      </div>
    </label>
  </div>,
]
`;
